src/components/Search/server-actions.tsx

```diff
@@ -52,6 +51,64 @@ export interface AskAnswerResult {
     hasAnswer: boolean;
 }
 
+export async function searchSiteContent(args: {
+    query: string;
+    siteSpaceIds?: string[];
+    cacheKey?: string;
+}): Promise<OrderedComputedResult[]> {
+    const { siteSpaceIds = [], query, cacheKey } = args;
+    const pointer = getContentPointer();
+
+    if ('siteId' in pointer && 'organizationId' in pointer) {
+        const searchResults = await api.searchSiteContent(
+            pointer.organizationId,
+            pointer.siteId,
+            query,
+            siteSpaceIds,
+            cacheKey,
+        );
+
+        // resolve all SiteSpaces so we can match up with the spaceId
+        const allSiteSpaces = await api.getSiteSpaces({
+            organizationId: pointer.organizationId,
+            siteId: pointer.siteId,
+            siteShareKey: pointer.siteShareKey,
+        });
+
+        if (siteSpaceIds.length === 0) {
+            // We are searching all of this Site's content
+
+            const processedSiteSpaces = allSiteSpaces.map((siteSpace) => {
+                // replace the published url for the "space" with the site's published url
+                siteSpace.space.urls.published = siteSpace.urls.published;
```

Comment:
Why?

Reply:
This is code ported over but it would seem we do not clear out the old published url for a space when migrating to sites. Usually it's the same, but since a space can belong to multiple sites it is always the site URL that we want. To support old published spaces and SiteSpaces it seems like this was done so they can coexist.

Reply:
It looks like it's due to `transformPageResult` accepting only a space for now, so we're looking for the `space.urls.published` value:
https://github.com/GitbookIO/gitbook/blob/aa8bbaddc46aa90ef30c970a057b329cd7a694ec/src/components/Search/server-actions.tsx#L186-L191

But we should adapt that for sites or even create a new transform function for them instead of manipulating the data to match the implementation for spaces I think.

Reply:
ok, made an update that pulls the space-specific logic into a function and common functionality into another.